src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,48 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(health = 100, name) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
+
+  die() {
+    Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, hidden, health = 100) {
+    super(hidden);
+    this.hidden = false;
+    this.health = health;
+    this.name = name;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health = 100, type = 'carnivore') {
+    super(health, name);
+    this.health = health;
```

Comment:
The 'health' property is already being set in the parent class constructor, so this line is redundant and can be removed.